src/recovery.rs

```diff
@@ -30,6 +30,10 @@ pub fn recover_partitions(keyspace: &Keyspace) -> crate::Result<()> {
         let partition_name = dirent.file_name();
         let partition_path = dirent.path();
 
+        if dirent.file_type()?.is_file() {
+            continue;
+        }
+
         assert!(
```

Comment:
There is already a check in place for checking whether the dirent is a file or not, the assert just needs to be changed to an if statement, and maybe change the assertion message ("found stray file ...") to a `log::warn`